icu4c/source/test/intltest/srchtest.cpp

```diff
@@ -2452,6 +2453,24 @@ void StringSearchTest::TestCoverage(){
     }
 }
 
+void StringSearchTest::TestBug22775() {
+    IcuTestErrorCode errorCode(*this, "TestBug22775()");
+    // Used to crash in Java due to bad management of the pattern collation element array.
+    UnicodeString pattern(
+        u"Xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+        u"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+        u"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+        u"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa響");
+    LocalPointer<RuleBasedCollator> collator(
+        dynamic_cast<RuleBasedCollator *>(
+            Collator::createInstance(Locale::getUS(), errorCode)),
+        errorCode);
+    collator->setAttribute(UCOL_STRENGTH, UCOL_PRIMARY, errorCode);
```

Comment:
Collator::createInstance could return nullptr and set errorCode, calling collator->setAttribute without checking errorCode is anti pattern in ICU

Reply:
done